### PR TITLE
Feature ta 3614

### PR DESCRIPTION
**Pull request by averysmithproductions**
for jira ticket https://us-sba.atlassian.net/browse/TA-3614

AC:
- [x] The D7 Event detail page is fixed

Before, now matter what event was clicked on, only the most recent event detail information displayed.

Now, any event that is clicked on from https://avery.ussba.io/events/ will open to its correct event detail information.

### CODE REVIEW

File: src/client/components/pages/event-page/event-page.jsx [hunk 1]
```diff
@@ -11,49 +11,35 @@ import clientConfig from '../../../services/client-config.js'
...
-          <Loader />
-        </div>
-      )
-    }
+    const { data, LOADING_STATE } = this.state
+
+    return (
+      <div>
+        {!eventId && <ErrorPage linkUrl="/events/find" linkMessage="find events page" />}
+        {eventId && LOADING_STATE === 'loaded' && (!isEmpty(data) ? <div><Event eventData={data} /></div> : <ErrorPage linkUrl="/events/find" linkMessage="find events page" />)}
```
Comment (amyd11) [inline src/client/components/pages/event-page/event-page.jsx]: This is a pretty long line with a bunch of logic. Do you think there's a way to make it a little more readable- potentially if statements?

Also, do we need that `div` around the `<Event eventData={data} />`? It doesn't seem like it's doing anything, we might be able to simplify this line by taking that out.

File: src/client/components/pages/event-page/event-page.jsx [hunk 2]
```diff
@@ -11,49 +11,35 @@ import clientConfig from '../../../services/client-config.js'
...
-    } else {
-      const { items } = await fetchSiteContent('events', queryArgs).catch(_ =>
-        this.setState({ data: null })
-      )
-      if (items && items[0]) {
-        this.setState({ data: items[0] })
-      }
-    }
+    const result = clientConfig.useD8EventsBackend ? results.hit : results.items
+    const data = !isEmpty(result) ? result[0] : null
```
Comment (amyd11) [inline src/client/components/pages/event-page/event-page.jsx]: nice refactor for this section of code

File: src/client/components/pages/event-page/event-page.jsx (hunk 1 above)
Comment (averysmithproductions) [reply src/client/components/pages/event-page/event-page.jsx]: updated!

File: src/client/components/pages/event-page/event-page.jsx [hunk 3]
```diff
@@ -36,8 +36,13 @@ class EventPage extends Component {
...
         {!eventId && <ErrorPage linkUrl="/events/find" linkMessage="find events page" />}
-        {eventId && LOADING_STATE === 'loaded' && (!isEmpty(data) ? <div><Event eventData={data} /></div> : <ErrorPage linkUrl="/events/find" linkMessage="find events page" />)}
-        {eventId && LOADING_STATE !== 'loaded' && <div className={styles.container}><Loader /></div>}
+        {eventId && <div>
+          {LOADING_STATE !== 'loaded' && <div className={styles.container}><Loader /></div>}
+          {LOADING_STATE === 'loaded' && <div>
+            {isEmpty(data) && <ErrorPage linkUrl="/events/find" linkMessage="find events page" />}
+            {!isEmpty(data) && <Event eventData={data} />}
+          </div>}
+        </div>}
```
Comment (averysmithproductions) [inline src/client/components/pages/event-page/event-page.jsx]: @amyd11 updated render return logic

**Pull request merged**